sensor/common/detector/detector.go

```diff
@@ -362,9 +362,9 @@ func (d *detectorImpl) runAuditLogEventDetector() {
 				return alerts[i].GetPolicy().GetId() < alerts[j].GetPolicy().GetId()
 			})
 			select {
-			case <-d.auditStopper.StopDone():
+			case <-d.auditStopper.Flow().StopRequested():
```

Comment:
[nit / rant / no immediate suggestion]

I'm not so happy with the pattern of nested selects with the same stop signal repeated. Like (pseudo-code):

```
select {
   <- stoppedPath:
         return
   <- workPath:
         select {
              <-stoppedPath:
                     return
              <- workPath:
                     // work
         }
}
```

I don't have a better suggestion right now. But maybe it would make sense to use non-blocking operations on the inner block and only handle stop signals at the top level. Since it's an infinite loop it will go back to the top level select anyway. It will still be a problem if the buffer becomes full then writing into it becomes blocking. One could make the argument that if the buffered `d.output` chan is full we probably have other serious problems.

Reply:
Note that I think this is required here as there are two sources of potential blocking:
1. `auditEvents := <-d.auditEventsChan`
2. `d.output <- &central.MsgFromSensor{`

We want to be sure to stop the goroutine in case of any of this places block, thus we are checking the stopper twice.

Reply:
That's exactly what I pointed out. My idea is having a **non-blocking** operation in the inner select.

Reply:
1. I agree that Go syntax in this scenario is clumsy. If it looked better, it might have annoyed less.
2. My _opinion_ is that this pattern is actually fine. I don't know of a more elegant way to express "read from here and write over there" async work in general case.
3. Introducing buffered channels may break things. I prefer to make mostly cosmetic changes than breaking things in this PR. Also don't want to fix too many things that I did not break.

Would you be ok to accept this unchanged?

Reply:
> Would you be ok to accept this unchanged?

Yes